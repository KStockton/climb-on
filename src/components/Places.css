.Places-bottom-container {
	background-color: pink;
	display: grid;
	grid-template-columns: 20% 60% 20%;
}
