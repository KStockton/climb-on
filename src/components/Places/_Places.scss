.Places {
	display: grid;
	grid-column-gap: 25px;
	grid-template-columns: 1fr 3fr;
	justify-items: center;
	margin: auto;
	max-width: 1200px;
}

.Places-header {
	color: #393939;
	font-size: 60px;
	grid-column-end: 3;
	grid-column-start: 1;
	margin: 0;
	padding: 20px 0;
	width: 100%;
}

.Places-container {
	width: 100%;
}