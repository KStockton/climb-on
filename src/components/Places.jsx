import React, { Component } from 'react';
import Sidebar from './SideBar.jsx';
import Place from './Place.jsx';

import './Places.scss';

class Places extends Component {

  render() {
    return (
      <section className="Places">
      <h2 className="Places-header">Results</h2>
        <Sidebar updateFilters={this.props.updateFilters}/>
        <section className="Places-container">
          {
            this.props.places.map((currentPlace, i) => {
              return ( 
                <Place
                  key={i}
                  place={currentPlace.place}
                  closestTown={currentPlace.closestTown}
                  climbingId={currentPlace.climbingId}
                  routes={currentPlace.routes}
                />
              )
            })
          }
        </section>
      </section>
    )
  }
}


export default Places;