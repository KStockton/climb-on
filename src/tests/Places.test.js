import React from 'react';
import Places from '../components/Places';
import { shallow } from 'enzyme';

const mockUpdateFilters = jest.fn();
const mockResults = [
  { 
    "place": "Golden Cliffs",
    "closestTown": "Golden, Colorado",
    "climbingId": 1,
    "routes": [
      { "climbingPlaceId": 1,
        "area": "Brown Cloud Crags",
        "routeName": "Louise",
        "difficultyLevel": "5.8",
        "type": ["sport"] },
      { "climbingPlaceId": 1,
        "area": "Brown Cloud Crags",
        "routeName": "Thelma",
        "difficultyLevel": "5.7",
        "type": ["sport"] }
    ]
  },
  { 
    "place": "Golden Cliffs",
    "closestTown": "Golden, Colorado",
    "climbingId": 1,
    "routes": [
      { "climbingPlaceId": 1,
        "area": "Brown Cloud Crags",
        "routeName": "Louise",
        "difficultyLevel": "5.8",
        "type": ["sport"] },
      { "climbingPlaceId": 1,
        "area": "Brown Cloud Crags",
        "routeName": "Thelma",
        "difficultyLevel": "5.7",
        "type": ["sport"] }
    ]
  }
]

describe('SearchForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Places 
        places={mockResults} 
        updateFilters={mockUpdateFilters}
      />
    )
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});