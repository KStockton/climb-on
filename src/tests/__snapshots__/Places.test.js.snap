// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Places should match the snapshot with all the data passed in 1`] = `
<section
  className="Places"
  id="Places"
>
  <h2
    className="Places-header"
  >
    Results
  </h2>
  <Sidebar
    updateFilters={[MockFunction]}
  />
  <section
    className="Places-container"
  >
    <Place
      climbingId={1}
      closestTown="Golden, Colorado"
      key="0"
      place="Golden Cliffs"
      routes={
        Array [
          Object {
            "area": "Brown Cloud Crags",
            "climbingPlaceId": 1,
            "difficultyLevel": "5.8",
            "routeName": "Louise",
            "type": Array [
              "sport",
            ],
          },
          Object {
            "area": "Brown Cloud Crags",
            "climbingPlaceId": 1,
            "difficultyLevel": "5.7",
            "routeName": "Thelma",
            "type": Array [
              "sport",
            ],
          },
        ]
      }
    />
    <Place
      climbingId={1}
      closestTown="Golden, Colorado"
      key="1"
      place="Golden Cliffs"
      routes={
        Array [
          Object {
            "area": "Brown Cloud Crags",
            "climbingPlaceId": 1,
            "difficultyLevel": "5.8",
            "routeName": "Louise",
            "type": Array [
              "sport",
            ],
          },
          Object {
            "area": "Brown Cloud Crags",
            "climbingPlaceId": 1,
            "difficultyLevel": "5.7",
            "routeName": "Thelma",
            "type": Array [
              "sport",
            ],
          },
        ]
      }
    />
  </section>
</section>
`;
